sklearn/utils/tests/test_validation.py

```diff
@@ -1213,3 +1214,60 @@ def test_check_sparse_pandas_sp_format(sp_format):
     assert sp.issparse(result)
     assert result.format == sp_format
     assert_allclose_dense_sparse(sp_mat, result)
+
+
+@pytest.mark.parametrize('ntype1, ntype2', [
+    ("longdouble", "float16"),
+    ("float16", "float32"),
+    ("float32", "double"),
+    ("int16", "int32"),
+    ("int32", "long"),
+    ("byte", "uint16"),
+    ("ushort", "uint32"),
+    ("uint32", "uint64"),
+    ("uint8", "int8"),
+])
+def test_check_pandas_sparse_invalid(ntype1, ntype2):
+    pd = pytest.importorskip("pandas", minversion="0.25.0")
+    df = pd.DataFrame({'col1': pd.arrays.SparseArray([0, 1, 0],
+                                                     dtype=ntype1),
+                       'col2': pd.arrays.SparseArray([1, 0, 1],
+                                                     dtype=ntype2)})
+
+    if parse_version(pd.__version__) < parse_version('1.1'):
+        with pytest.raises(ValueError,
+                           match="Pandas DataFrame with mixed "
+                                 "sparse extension arrays"):
+            check_array(df, **{'accept_sparse': ['csr', 'csc'],
+                               'ensure_min_features': 2})
+    else:
```

Comment:
The else statement is for the case when pandas version >= 1.1. In this case the error is not raised because pandas fixed the issue.
To summarize. 
a. pandas < 0.25
    We skip the whole test. Here the sparse DataFrame will not return a sparse matrix. It will return a numpy array. So there is nothing to test.
b. 0.25 <= pandas < 1.1
    The sparse DataFrame returns a sparse coo_matrix.  For some combinations of numeric types, the sparse coo_matrix is of dtype Object. When trying to obtain a csr_matrix from it, scipy raises an error. This is the issue we are handling.
c. pandas >= 1.1
   Pandas fixed the issue with coo_matrix of dtype object
   The sparse DataFrame returns a valid sparse coo_matrix always (not of dtype object), so we will always get a csr_matrix from it. So here no error will be raised.
   This is what the "else" in the test verfies. I will add a coment  explaining this.

Reply:
Good thanks